src/thinblock.cpp

```diff
@@ -378,12 +378,15 @@ bool CXThinBlock::process(CNode *pfrom,
         }
     } // End locking cs_orphancache, mempool.cs and cs_xval
 
-    // This must be checked outside the above section or a deadlock may occur
+    // This must be done outside of the above section or a deadlock may occur.
     if (!fMerkleRootCorrect)
     {
-        LOCK(cs_main);
-        Misbehaving(pfrom->GetId(), 100);
-        return error("Thinblock merkelroot does not match computed merkleroot, peer=%d", pfrom->GetId());
+        vector<CInv> vGetData;
+        vGetData.push_back(CInv(MSG_THINBLOCK, header.GetHash()));
+        pfrom->PushMessage("getdata", vGetData);
```

Comment:
Please use NetMsgType::GETDATA here (and at the other place in this file).

I approve with that change; and I think this should be considered for backport to release as it solves the banning DoS of #485 until we solve the collision issue.